code/game/objects/items/storage/storage.dm

```diff
@@ -506,7 +540,7 @@
 	var/turf/T = get_turf(src)
 	hide_from(usr)
 	for(var/obj/item/I in contents)
-		remove_from_storage(I, T)
+		remove_from_storage(I, T, usr)
```

Comment:
Maybe out of scope of this pr, but every quick_empty call is done in a proc where you have access to mob/user

You could replace all these usr by user

Reply:
I'll do this in my complete storage overhaul/refactor not this PR